robot/framesystem/framesystem.go

```diff
@@ -59,33 +70,31 @@ type InputEnabled interface {
 //
 //	// Transform the first point cloud in the list from its reference frame to the frame of 'myArm'.
 //	transformed, err := fsService.TransformPointCloud(context.Background(), pointClouds[0], referenceframe.World, "myArm")
-//
-// CurrentInputs example:
-//
-//	myCurrentInputs, err := fsService.CurrentInputs(context.Background())
-//
-//	frameSystem, err := fsService.FrameSystem(context.Background(), nil)
-type Service interface {
-	resource.Resource
+type RobotFrameSystem interface {
+	// FrameSystemConfig returns the individual parts that make up a robot's frame system
+	FrameSystemConfig(ctx context.Context) (*Config, error)
+
+	// GetPose returns the pose a component within a frame system.
+	// It returns a `PoseInFrame` describing the pose of the specified component relative to the specified destination frame.
+	// The `supplemental_transforms` argument can be used to augment the machine's existing frame system with additional frames.
+	GetPose(
+		ctx context.Context,
+		componentName, destinationFrame string,
+		supplementalTransforms []*referenceframe.LinkInFrame,
+		extra map[string]interface{},
+	) (*referenceframe.PoseInFrame, error)
 
 	// TransformPose returns a transformed pose in the destination reference frame.
 	// This method converts a given source pose from one reference frame to a specified destination frame.
 	TransformPose(
 		ctx context.Context,
 		pose *referenceframe.PoseInFrame,
 		dst string,
-		additionalTransforms []*referenceframe.LinkInFrame,
+		supplementalTransforms []*referenceframe.LinkInFrame,
```

Comment:
Should this also take an `extra`?

Reply:
No I asked about that it was determined out of scope in the scoping